pkg/api/v1/routes/handlers.go

```diff
@@ -266,8 +278,23 @@ func (r *Routes) serverConditionCreate(c *gin.Context) (int, *v1types.ServerResp
 	}
 
 	// XXX: this operation can't ignore errors
+	// - if this method fails, should the condition be deleted?
```

Comment:
I think failing the condition is better here. That way we don't leave a caller confused as to why the condition they might have just seen (depending on how the UI/API interacts with serverservice) is now gone.

Reply:
Sounds right, will address this in another PR.